website/content/api-docs/volumes.mdx

```diff
@@ -782,8 +792,335 @@ $ curl \
+        "AccessMode": "single-node-writer",
+        "AttachmentMode": "file-system"
+      },
+      {
+        "AccessMode": "single-node-writer",
+        "AttachmentMode": "block-device"
+      }
+    ],
+    "Constraints": [
+      {
+        "LTarget": "${attr.kernel.name}",
+        "RTarget": "linux",
+        "Operand": "="
+      }
+    ],
+    "Parameters": {
+      "foo": "bar"
+    }
+  }
+}
+```
+
+### Sample Request
+
+```shell-session
+$ curl \
+    --request PUT \
+    --data @payload.json \
+    https://localhost:4646/v1/volume/host/create
+```
+
+### Sample Response
+
+```json
+{
+  "Volume": {
+    "Allocations": [],
+    "CapacityBytes": 21474836480,
+    "Constraints": [
+      {
+        "LTarget": "${attr.kernel.name}",
+        "RTarget": "linux",
+        "Operand": "="
+      }
+    ],
+    "CreateIndex": 11,
+    "CreateTime": 1736191993011594200,
+    "ID": "c0f7ee7d-5cc6-92fd-f2b5-14b79f01979f",
+    "ModifyIndex": 11,
+    "ModifyTime": 1736191993011594200,
+    "Name": "example",
+    "Namespace": "default",
+    "NodeID": "45460554-cc67-11ef-84b7-33d383a55487",
+    "NodePool": "prod",
+    "Parameters": {
+      "foo": "bar"
+    },
+    "PluginID": "mkdir",
+    "RequestedCapabilities": [
+      {
+        "AccessMode": "single-node-writer",
+        "AttachmentMode": "file-system"
+      },
+      {
+        "AccessMode": "single-node-writer",
+        "AttachmentMode": "block-device"
+      }
+    ],
+    "RequestedCapacityMaxBytes": 21474836480,
+    "RequestedCapacityMinBytes": 10737418240,
+    "State": "pending"
+  },
+  "Warnings": null
+}
+```
+
+## Register Dynamic Host Volume
+
+This endpoint registers an already-created Dynamic Host Volume with Nomad.
+
+| Method | Path                       | Produces           |
+|--------|----------------------------|--------------------|
+| `PUT`  | `/v1/volume/host/register` | `application/json` |
+
+The table below shows this endpoint's support for [blocking queries][] and
+[required ACLs][].
+
+| Blocking Queries | ACL Required                                                                                                                     |
+|------------------|----------------------------------------------------------------------------------------------------------------------------------|
+| `NO`             | `namespace:host-volume-register` or<br/>`namespace:host-volume-write`.<br/>`namespace:sentinel-override` if `PolicyOverride` set |
+
+### Parameters
+
+- `Volume` `(Volume: <required>)` - Specifies the JSON definition of the host
+  volume. You should include the ID field if you are updating an existing
+  volume.
+
+- `PolicyOverride` `(bool: false)` - If set, any soft mandatory Sentinel
+  policies will be overridden. This allows a volume to be registered when it
+  would be denied by policy.
+
+### Sample Payload
+
+```json
+{
+  "PolicyOverride": false,
+  "Volume": {
+    "Name": "example",
+    "Namespace": "default",
+    "NodePool": "prod",
```

Comment:
hadn't thought about this before -- should we require `NodeID` for DHV `register`?  if a vol already exists, it must exist on a specific node, eh?

I suppose if someone pre-populates a volume across a whole lot of nodes (in a pool, say), then `register` could be run repeatedly without a `NodeID`, until it starts to error that no nodes meet constraints...

Reply:
https://github.com/hashicorp/nomad/pull/24795